### DISCUSSION TITLE
FeO data

### DISCUSSION THREAD

**Discussion by petrvokac** [pycalphad/pycalphad#411] [General]
Hello,

I found intersing analysis of FeO system:

https://www.researchgate.net/publication/270008110_Thermodynamic_Reevaluation_of_the_Fe-O_System?enrichId=rgreq-b87794d797d2dbc5ffd733238945318b-XXX&enrichSource=Y292ZXJQYWdlOzI3MDAwODExMDtBUzozMTU0MTcwNjE0NjIwMTZAMTQ1MjIxMjY5NTIwMQ%3D%3D&el=1_x_2&_esc=publicationCoverPdf

It provides supplementary data:

https://ars.els-cdn.com/content/image/1-s2.0-S0364591614001023-mmc1.zip

However, this file is not in TDB format.

Can it be read by pycalphad or can it be somehow converted to TDB?

Thank you,

Petr

**Comment by richardotis** [comment]
The development version of pycalphad provides initial support for reading DAT files (since https://github.com/pycalphad/pycalphad/pull/389 was merged) but we haven't released a new version that has the feature yet. The next release will have support.

However, it looks like our current DAT support is not quite sufficient for reading this file (as of pycalphad develop [f8e3951](https://github.com/pycalphad/pycalphad/commit/f8e39510ffe282ba01171c79a889f053ce7a2811)):
```python
import requests, zipfile
from io import BytesIO
r = requests.get('https://ars.els-cdn.com/content/image/1-s2.0-S0364591614001023-mmc1.zip', stream=True)
z = zipfile.ZipFile(BytesIO(r.content))
file = z.open('mmc1.dat').read().decode('utf-8')
from pycalphad import Database
dbf = Database.from_string(file, fmt='dat')
```
```python
NotImplementedError: phase type QKTOM not yet supported
```
@bocklund can chime in with any relevant issues to track about this feature.

**Reply by bocklund** [reply]
I need to check the database, but this shouldn't be too difficult to support. This seems to be a magnetic QKTO model. We have DAT support for reading QKTO models and magnetic parameters for Redlich-Kister models in the DAT.

This might be as simple as adding `QKTOM` to https://github.com/pycalphad/pycalphad/blob/f8e39510ffe282ba01171c79a889f053ce7a2811/pycalphad/io/cs_dat.py#L1120

**Reply by richardotis** [reply]
That change isn't quite sufficient on its own; then you get a `NotImplementedError: phase type 1 not yet supported`
